src/compiler/program.ts

```diff
@@ -393,12 +394,12 @@ namespace ts {
                 // If '--lib' is not specified, include default library file according to '--target'
                 // otherwise, using options specified in '--lib' instead of '--target' default library file
                 if (!options.lib) {
-                    processRootFile(host.getDefaultLibFileName(options), /*isDefaultLib*/ true);
+                    processRootFile(host.getDefaultLibFileName(options), /*isDefaultLib*/ true, /*ignoreNoDefaultLib*/ false);
                 }
                 else {
-                    const libDirectory = host.getDefaultLibLocation ? host.getDefaultLibLocation() : getDirectoryPath(host.getDefaultLibFileName(options));
+                    const libDirectory = getLibDirectory();

```

Comment:
this shadows the outer `libDirectory`. Can you change it so that the code is less confusing?